sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -245,6 +245,7 @@ datetimeConstantLiteral
     | CURRENT_TIME
     | CURRENT_TIMESTAMP
     | DAY_OF_YEAR
+    | HOUR_OF_DAY
```

Comment:
`datetimeConstantLiteral` is a function which could be called without `()` too, e.g. 
```sql
select HOUR_OF_DAY, HOUR_OF_DAY();
```
`HOUR_OF_DAY` should be in this list. `DAY_OF_YEAR`, `WEEK_OF_YEAR` and `MONTH_OF_YEAR` too, actually. See `dateTimeFunctionName`.

Reply:
> `HOUR_OF_DAY` should be in this list. `DAY_OF_YEAR`, `WEEK_OF_YEAR` and `MONTH_OF_YEAR` too, actually. See dateTimeFunctionName.

They are currently in this list. Did you mean to say that they _**shouldn't**_ be in this list?

Reply:
Right, sorry. My mistake, but you understand everything correct.

Reply:
No worries. Thanks Yury! Have a good vacation!

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/195/commits/4255ec9e0389abb364baa71cdd4ce10495d6be99